CMakeLists.txt

```diff
@@ -273,7 +273,11 @@ project (Jerry C ASM)
   add_jerry_compile_warnings(all extra format-nonliteral init-self conversion sign-conversion format-security missing-declarations pedantic)
   add_jerry_compile_flags(-Wno-stack-protector -Wno-attributes)
   if(CMAKE_COMPILER_IS_GNUCC)
-    add_jerry_compile_warnings(logical-op)
+    if(${USE_JERRY_LIBC})
+      add_jerry_compile_flags(-Werror -Wlogical-op)
+    else()
+      add_jerry_compile_warnings(logical-op)
+    endif()
```

Comment:
Why the if-else construct? Cannot we just leave `add_jerry_compile_warnings(logical-op)` in place and add
```
if(${USE_JERRY_LIBC})
 add_jerry_compile_flags(-Werror)
else()
```